File: src/Coords/Waypoint.java
package Coords;

public class Waypoint {
    public double x,dx,ddx,dddx;
    public double y,dy,ddy,dddy;

    //for qhs
    public Waypoint(double x,double y,double dx,double dy,double ddx,double ddy) {
        this.x=x;
        this.y=y;
        this.dx=dx;
        this.dy=dy;
        this.ddx=ddx;
        this.ddy=ddy;
    }

    //for chs
    public Waypoint(double x, double y, double dx, double dy) {
        this.x=x;
        this.y=y;
        this.dx=dx;
        this.dy=dy;
    }
}

File: src/Coords/Waypoint.kt
package Coords

open class Waypoint//for qhs
(x: Double, y: Double, var dx: Double, var dy: Double, var ddx: Double, var ddy: Double) : Point(x, y) {
    var dddx: Double = 0.toDouble()
    var dddy: Double = 0.toDouble()

    fun unit() : Waypoint {
        val m = Point(dx, dy).magnitude
        val dm = Point(ddx, ddy).magnitude
        return Waypoint(x, y, dx/m, dy/m, ddx/dm, ddy/dm)
    }

    override fun toString(): String {
        return "($x, $y, $dx, $dy)";
    }
}
